KLSadd_insertion/src/updateChapters.py

```diff
@@ -114,48 +331,67 @@ def findReferences(chapter):
+                ref14_3.append(index)
+            for unit in math_people:
+                subunit = unit[unit.find(" ")+1: unit.find("#")]
                 # System of checks that verifies if section is in chapter
-                if ((w in subunit) or (ws in subunit)) and (chaptercheck in unit) and (len(w) == len(
-                        subunit)) or (("Pseudo Jacobi" in w) and ("Pseudo Jacobi (or Routh-Romanovski)" in subunit)):
-                    canAdd = True
+                if (w in subunit or ws in subunit) and (chaptercheck in unit) and len(w) == len(subunit) or\
+                ("Pseudo Jacobi" in w and "Pseudo Jacobi (or Routh-Romanovski)" in subunit):
+                    canadd = True
                     if chapticker == 0:
-                        ref9II.append(index)
-                        ref9III.append(index)
+                        ref9_3.append(index)
                     elif chapticker == 1:
-                        ref14II.append(index)
-                        ref14III.append(index)
-        if("\\subsection*{References}" in word) and (canAdd):
+                        ref14_3.append(index)
+                    special_detector = 0
+        if "\\subsection*{References}" in word and canadd:
             # Appends valid locations
             references.append(index)
             if chapticker == 0:
-                ref9II.append(index)
-                ref9III.append(index)
+                ref9_3.append(index)
             elif chapticker == 1:
-                ref14II.append(index)
-                ref14III.append(index)
-            canAdd = False
-        if ("subsection*{" in word and "References" not in word):
+                ref14_3.append(index)
+            canadd = False
+        if "subsection*{" in word and "References" not in word:
             if chapticker == 0:
-                ref9III.append(index)
+                ref9_3.append(index)
             elif chapticker == 1:
-                ref14III.append(index)
-    print(ref9II)
-    print (ref14II)
-    print(ref9III)
-    print(ref14III)
-    return references
+                ref14_3.append(index)
+        if "\\section{" in word and special_detector == 1 and chaptercheck == "14":
+            w2 = word[word.find("{") + 1: word.find("}")]
+            if "Bessel" not in w2:
+                ref14_3.append(index)
+
+    bqlegendrecheck = 2
+
+    # Special lines that the program does not normally register
+    if bqlegendrecheck > 0:
+        pass
 
+    return references, ref9_3, ref14_3
 
-def referencePlacer(chap, references, p, kls, refII, refIII, specref):
+
+def reference_placer(chap, references, p, chapticker2):
+    """
+    Places identified additions from the addendum into the chapter.
+
+    :param chap: Chapter receiving the additions.
+    :param references: List containing references
```

Comment:
references to what? don't bother with comments if they're so incredibly obvious as this one is. Make the comment actually useful